### DISCUSSION TITLE
Is there a way to test for viewport overflow in Cypress? (Also why doesn't scrollTo(x) work?)

### DISCUSSION THREAD

**Discussion by LandGod** [cypress-io/cypress#15663] [Questions and Help]
I've been trying to rig up a test for detecting if any of the elements on a given page are overflowing the viewport, but I'm running into a ton of roadblocks. I'm not sure if this is a case of me just misunderstanding functionality, an actual bug in Cypress, or a functionality that is simply not implemented in Cypress.

I'm running Cypress version `6.8.0`
Here is the code that I've tried so far:
<details>

```
    cy.viewport("iphone-5");

    cy.visit("/");
    // Override Cypress's default behavior of hiding overflow-x on body.
    cy.document().then((document) => {
      document.getElementsByTagName("body")[0].style.overflowX = "auto";
    });
    cy.wait(1000)
    // Scroll to bottom of page (to ensure all elements actually start to load up)
    // Then attempt to scroll sideways for a tenth of a second and asserts that no horizontal scroll actually occurred.
    cy.scrollTo("bottom");
    cy.wait(2000);
    cy.scrollTo("bottomRight").then((window) => {
      expect(window.scrollX).to.equal(0);
    });
```

</details>

I've tried a number of iterations of trying to scroll horizontally but nothing ever happens. No matter what argument I give to `cy.scrollTo` it never does any horizontal scrolling and then of course it will happily report that window.scrollX is 0 and report success despite not actually doing anything. 

As you can see I also tried adding some `wait`s to make sure I wasn't having async issues, but that doesn't seem to make any difference either.

#### So here are my two questions:
1. Why isn't this working?
2. Is there a better way to test for viewport overflow that I should use instead? 

Thank you.

PS: I can see the horizontal scrollbar on the window after setting `overflow-x: auto` on the body, so I know that the scroll is possible and I can even do with my mouse when running in dashboard mode.

<details>

![Screenshot 2021-03-25 133221](https://user-images.githubusercontent.com/8890238/112542617-04b9b680-8d72-11eb-983c-83bf8e3254da.png)

</details>

**Comment by andolf** [comment]
Reviving this. I'm having the same trouble with scrolling `window` when the body is set to `overflow: auto`. Removing it fixes the issue but it needs to be there to prevent scrolling when displaying a `<dialog>` element with non-scrollable backdrop.
